src/platform/packages/shared/kbn-unified-data-table/src/hooks/use_sorting.ts

```diff
@@ -113,3 +115,31 @@ export const useSorting = ({
 
   return { sortedRows, sorting };
 };
+
+export const isSortable = ({
+  isPlainRecord,
+  columnName,
+  columnSchema,
+  dataViewField,
+}: {
+  isPlainRecord: boolean | undefined;
+  columnName: string;
+  columnSchema: string;
+  dataViewField: DataViewField | undefined;
+}): boolean => {
+  if (isPlainRecord) {
+    // TODO: would be great to have something like `sortable` flag for text based columns too
+    if (columnName === SOURCE_COLUMN) {
+      return false; // _source column is not sortable
+    }
+    return Boolean(
+      columnSchema !== kibanaJSON ||
+        (dataViewField?.type &&
+          Object.values(NonStringSortableFieldType).includes(
+            dataViewField.type as NonStringSortableFieldType
+          ))
+    );
```

Comment:
@stratoula Would you advise to exclude only geo fields instead? Can't decide if it should check for any json-like fields excluding the ones which are handled separately. Or it should exclude only geo fields explicitly. What would be safer? We used to disable sorting only for `_source` in ES|QL mode.